std/file.d

```diff
@@ -4293,11 +4293,6 @@ string tempDir() @trusted
             DWORD len = GetTempPathW(buf.length, buf.ptr);
             if (len) cache = buf[0 .. len].to!string;
         }
-        else version(Android)
-        {
-            // Don't check for a global temporary directory as
-            // Android doesn't have one.
-        }
```

Comment:
Added another Android commit, as ldc is now packaged on Android itself, and the Termux app does set `TMPDIR`.  This was versioned out because it was assumed that most would run D code from their own Android app, ie an apk, which don't have any global temp directory.  However, now that D can be used from the command-line on Android itself, with an app that sets `TMPDIR` for all command-line D tools, just use the below Posix logic.  You will have to set [the tempDir by hand for each apk anyway](https://gist.github.com/joakim-noah/09a59bdd6e04dd2f3d3ad6e15573ad97#file-phobos_1-3_ldc_arm-L9), if you use it. 

This already hit me when building [`rdmd` for Android, where I had to patch it to check `TMPDIR`](https://github.com/termux/termux-packages/pull/1377/files#diff-267b88bd99296808606d2a526c27d831), can take that out with this change.